db/recovery_db.go

```diff
@@ -157,12 +156,61 @@ func (db *RecoveryDB) GetSharedSwapPrivateKey(id types.Hash) (*mcrypto.PrivateSp
 	return mcrypto.NewPrivateSpendKeyFromHex(skHex)
 }
 
+type xmrmakerKeys struct {
+	PublicSpendKey string
+	PrivateViewKey string
+}
+
+// PutXMRMakerSwapKeys is called by the xmrtaker to store the counterparty's swap keys.
+func (db *RecoveryDB) PutXMRMakerSwapKeys(id types.Hash, sk *mcrypto.PublicKey, vk *mcrypto.PrivateViewKey) error {
+	val, err := json.Marshal(&xmrmakerKeys{
+		PublicSpendKey: sk.Hex(),
+		PrivateViewKey: vk.Hex(),
+	})
+	if err != nil {
+		return err
+	}
+
+	key := getRecoveryDBKey(id, xmrmakerKeysPrefix)
+	return db.db.Put(key[:], val)
+}
+
+// GetXMRMakerSwapKeys is called by the xmrtaker during recovery to retrieve the counterparty's
+// swap keys.
+func (db *RecoveryDB) GetXMRMakerSwapKeys(id types.Hash) (*mcrypto.PublicKey, *mcrypto.PrivateViewKey, error) {
+	key := getRecoveryDBKey(id, xmrmakerKeysPrefix)
+	value, err := db.db.Get(key[:])
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var info xmrmakerKeys
+	err = json.Unmarshal(value, &info)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sk, err := mcrypto.NewPublicKeyFromHex(info.PublicSpendKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	vk, err := mcrypto.NewPrivateViewKeyFromHex(info.PrivateViewKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return sk, vk, nil
+}
+
 // DeleteSwap deletes all recovery info from the db for the given swap.
 func (db *RecoveryDB) DeleteSwap(id types.Hash) error {
 	keys := [][]byte{
+		getRecoveryDBKey(id, relayerInfoPrefix),
```

Comment:
I'm not opposed to leaving the Recovery information separate from the Info information, but it feels like we can end up with dangling recovery entries (recovery entries that point to a Info entry that doesn't exist). And we'll never delete those entries if they are dangling.  Can that happen?

Reply:
I don't think can happen as `swap.Info` isn't ever deleted from the database, only recovery info is.